src/cider/nrepl/middleware/debug.clj

```diff
@@ -438,12 +438,28 @@
             (eval form)
             (alter-meta! v assoc ::instrumented instrumented)))))))
 
+(def step-in-blacklist
+  "A set of names or regexes specifying namespaces that should not be stepped in
+  to. By default, this contains only \"clojure.core\"."
+  (atom #{"clojure.core"}))
```

Comment:
> But in light of your earlier comment, do you think I should just switch back to a simpler implementation that hard-codes a list of namespaces to skip (initially containing only "clojure.core")?

I think so. And I think `breakpoint-if-interesting` should also avoid instrumenting if the current `*ns*` is in  this list.